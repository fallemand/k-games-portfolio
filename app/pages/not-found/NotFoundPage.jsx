import React from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import './_not-found.scss';

const NotFoundPage = ({ history }) => {
  const handleClick = () => {
    history.push('/search');
  };

  return (
    <div className="not-found">
      <h1 className="not-found__title">404</h1>
      <h2 className="not-found__subtitle">{"We couldn't find the page you're looking for :(".toString()}</h2>
      <button
        className="not-found__button"
        type="button"
        onClick={handleClick}
      >
        Go Home!
      </button>
    </div>
  );
};
NotFoundPage.propTypes = {
  history: PropTypes.object.isRequired,
};

export default withRouter(NotFoundPage);
