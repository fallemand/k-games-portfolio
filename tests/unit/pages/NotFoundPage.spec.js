import React from 'react';
import NotFoundPage from '../../../app/pages/not-found/NotFoundPage';

describe('NotFoundPage', () => {
  const props = {
    history: { push: jest.fn() },
  };

  it('render', () => {
    const component = shallow(<NotFoundPage {...props} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('click on button should redirect to search', () => {
    const component = shallow(<NotFoundPage {...props} />);
    component.find('button').simulate('click');
    expect(props.history.push).toHaveBeenCalledWith('/search');
  });
});
