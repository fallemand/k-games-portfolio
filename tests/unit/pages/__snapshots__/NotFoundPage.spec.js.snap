// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage render 1`] = `
"<div className='not-found'>
  <h1 className='not-found__title'>
    404
  </h1>
  <h2 className='not-found__subtitle'>
    We couldn&#39;t find the page you&#39;re looking for :(
  </h2>
  <button className='not-found__button' type='button' onClick={[Function: handleClick]}>
    Go Home!
  </button>
</div>"
`;
